builder/builder.go

```diff
@@ -149,7 +150,9 @@ func (b *Builder) RunTask(ctx context.Context, task *graph.Task) error {
 	var deps []*image.Dependencies
 	for _, step := range task.Steps {
 		log.Printf("Step ID: %v marked as %v (elapsed time in seconds: %f)\n", step.ID, step.StepStatus, step.EndTime.Sub(step.StartTime).Seconds())
-
+		if step.UseBuildCacheForBuildStep() && runtime.GOOS == util.LinuxOS {
```

Comment:
1. Can you add some comments to explain the limitation?
2. Also why add an additional runtime os check? does it work for windows?

Reply:
no - but if its a Windows Task, and if the task file contains "cache: enabled"

then `step.UseBuildCacheForBuildStep()` will return true. However, since it is Windows - we do not error out. We fall back to normal build.

So in this scenario, if its windows - then we will check for dep anyway.

Reply:
I forget we fallback to docker build. Do we print out any warning to tell user `cache:enabled` doesn't work on windows?

Reply:
yes, https://github.com/Azure/acr-builder/blob/master/graph/task.go#L327